packages/cyberstorm/src/components/Forms/CreateTeamForm/CreateTeamForm.tsx

```diff
@@ -0,0 +1,111 @@
+"use client";
+import { faArrowsRotate } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { useController, useForm, Form } from "react-hook-form";
+import * as Button from "../../Button";
+import { TextInput } from "../../TextInput/TextInput";
+import { useToast } from "../../Toast/Provider";
+import styles from "./CreateTeamForm.module.css";
+import { z } from "zod";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { isRecord } from "../../../utils/type_guards";
+
+interface errorResponse {
+  error: { message: string };
+}
+
+function isErrorResponse(response: unknown): response is errorResponse {
+  return (
+    isRecord(response) &&
+    isRecord(response.error) &&
+    typeof response.error.message === "string"
+  );
+}
+
+export function CreateTeamForm() {
+  const toast = useToast();
+
+  interface formFields {
```

Comment:
* By convention it should be FormFields with capital F
  * But to stick to what RHF's own documentation is using, it could be called `Schema`
* The interface and the schema below should probably be defined outside the function?
* If you check the [example in the documentations](https://react-hook-form.com/docs/useform#resolver) you'll find that you can infer the interface instead of manually duplicating it. I think this issue is borderline blocker.